core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -674,13 +596,9 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
    * elector
    */
   def startup() = {
-    inLock(controllerContext.controllerLock) {
-      info("Controller starting up")
-      registerSessionExpirationListener()
-      isRunning = true
-      controllerElector.startup
-      info("Controller startup complete")
-    }
+    controllerEventQueue.put(Startup)
+    controllerEventQueue.put(Elect)
```

Comment:
Do we need to do this through the queue? It seems the event thread could just do these two steps at the beginning.